### ISSUE TITLE
[mu4e bug] installation fails when infodir is absolute

### ISSUE THREAD

**Issue by jian-lin** [djcb/mu#2548]
**Describe the bug**

I attempt to change `mu` in Nixpkgs to a multiple-output package:
- `mu4e` related files are put into /nix/store/hash1-mu-version-mu4e.  The expected result:
```
/nix/store/hash1-mu-version-mu4e
└── share
    ├── emacs
    │   └── site-lisp
    │       └── elpa
    │           └── mu4e
    │               ├── mu4e-actions.el
    │               ├── mu4e-actions.elc
    │               ├── other files ...
    └── info
        ├── dir
        └── mu4e.info.gz
```
- other files are put into /nix/store/hash2-mu-version.  The expect result:
```
/nix/store/hash2-mu-version
├── bin
│   └── mu
└── share
    ├── doc
    │   └── mu
    │       ├── mu4e-about.org
    │       └── NEWS.org
    └── man
        ├── man1
        │   ├── mu.1.gz
        │   ├── other files ...
        ├── man5
        │   └── mu-bookmarks.5.gz
        └── man7
            ├── mu-easy.7.gz
            └── mu-query.7.gz
```

So I set --infodir=/nix/store/hash1-mu-version-mu4e/share/info and the installation fails

```console
Running custom install script '/bin/sh /build/source/build-aux/meson-install-info.sh share/info mu4e.info'
install-info: /nix/store/ixn5rjddm29y7rdqg7045j8kzb1ijv25-mu-1.10.7/share/info/dir: could not read (No such file or directory) and could not create (No such file or directory)
gzip: /nix/store/ixn5rjddm29y7rdqg7045j8kzb1ijv25-mu-1.10.7/share/info/mu4e.info: No such file or directory
FAILED: install script '/bin/sh /build/source/build-aux/meson-install-info.sh share/info mu4e.info' exit code 1, stopped
FAILED: meson-internal__install
```

It seems `infodir` is not respected [here](https://github.com/djcb/mu/blob/4d3d94d7b156358bf0fff29bf1fec61870024514/mu4e/meson.build#L154-L156).

I see some related discussion in https://github.com/djcb/mu/issues/2285.  Maybe @eli-schwartz has some ideas.

**How to Reproduce**
Set infodir to an absolute path

**Environment**
mu: 1.10.7

**Checklist**
- [X] you are running either an 1.8.x/1.10.x release or `master` (otherwise please upgrade)
- [X] you can reproduce the problem without 3rd party extensions (including Doom/Evil, various extensions etc.)
- [X] you have read all of the above

Please make sure you all items in the checklist are set/met before filing the ticket. Thank you!

**Comment by eli-schwartz** [comment]
> I see some related discussion in #2285. Maybe @eli-schwartz has some ideas.

I pointed out in that ticket that it should be run with the argument `get_option('prefix') / get_option('infodir'),` which is generally necessary in order to respect custom --infodir options.

This is as opposed to the bug you are hitting, which is that it hardcodes `share/info`.

The solution isn't complicated, the advice I pointed out there still applies.
